src/inmanta/data/model.py

```diff
@@ -338,3 +338,30 @@ def __init__(
         self.end = end
         self.first_id = first_id
         self.last_id = last_id
+
+
+class ResourceDetails(BaseModel):
+    """The details of a released resource
+    :param resource_id: The id of the resource
+    :param resource_type: The type of the resource
+    :param agent: The agent associated with this resource
+    :param attribute: The name of the identifying attribute of the resource
+    :param resource_id_value: The value of the identifying attribute of the resource
+    :param last_deploy_finished: The time the last deploy of the resource was finished
```

Comment:
could you add an example to the testcase where
- the resource never goes to deployed, skipped or failure (i.e. no deploy was ever finished)
- the resource only goes to deployed in a version that is not the latest
- the resource only goes to deployed in a version that is prior to first (i.e. it was never deployed with this hash)

Reply:
(the definition given here may be one of the hardest ones to implement, perhaps it would be easier to relax this definition a bit)

Reply:
Added the test cases.
About the `last_deploy`, I changed the docstring to reflect what it is more accurately. 
I don't know if this is the most useful timestamp a user can get about the resource.